c-wrapper/c-examples/challenge_response/Makefile

```diff
@@ -0,0 +1,20 @@
+# Copyright 2022 Contributors to the Veraison project.
+# SPDX-License-Identifier: Apache-2.0
+
+default: run
+
+# If the header has not been generated, then we need to 'cargo build' the root workspace
+../../veraison_client_wrapper.h:
+	pushd ../../.. && cargo build && popd
+
+challenge_response.o: challenge_response.c ../../veraison_client_wrapper.h
+	gcc -Wall -c -I../.. challenge_response.c
+
+challenge-response: challenge_response.o
+	gcc -Wall -o challenge-response challenge_response.o  ../../../target/debug/libveraison_apiclient_ffi.a -framework Security -framework CoreFoundation -liconv -lSystem -lresolv -lc -lm -liconv
```

Comment:
Yes, it should be multi-platform, and you caught me in the act of unthinkingly shovelling some MacOS stuff in there because I happened to be developing on a MacOS machine at the time. If the standard practice these days is still to have a per-platform Makefile, then I would be happy to rename this one and perhaps also introduce a Makefile.linux. Let me look into that before we merge this.

Reply:
👍 

For an up-to-date list of implicit GNU make variables (`CFLAGS`, `LDFLAGS`, `LDLIBS` may be applicable to our case) see [here](https://www.gnu.org/software/make/manual/html_node/Implicit-Variables.html)

You could put the (parametrised) rules in a build.mk and have a Makefile.<$OS> that sets the interface variables and `include build.mk`.